ArduPlane/quadplane.cpp

```diff
@@ -3708,7 +3708,9 @@ bool QuadPlane::show_vtol_view() const
     bool show_vtol = in_vtol_mode();
 
     if (tailsitter.enabled()) {
-        if (show_vtol && (transition_state == TRANSITION_ANGLE_WAIT_VTOL)) {
+        if (show_vtol &&
+            (transition_state == TRANSITION_ANGLE_WAIT_VTOL) &&
+            hal.util->get_soft_armed()) {
```

Comment:
This should not make any difference because transitions complete instantly when disarmed, so you should never* be in `TRANSITION_ANGLE_WAIT_VTOL` while disarmed.

*There is race condition here, so you might for less than one loop.

Reply:
it did matter while doing an ahrstrim or accelcal. In that case the main loop stops for enough time for the angle wait timeout to trigger. I made this change so that the HUD doesn't go crazy while doing an ahrstrim

Reply:
That should only happen if you change modes from FW to VTOL during the calibration?

I think my preferred fix would be to check if trim or cal is running and in that case do a early return on 3709, with a nice comment explaining the loop is stuck. That would keep the stuff that matters for flying clearly separate from this new case.

Reply:
We have `ins.calibrating()` looks like there is no state for trim

Reply:
> That should only happen if you change modes from FW to VTOL during the calibration?

no, it happens with no mode switch. Just fire up the HWing sim in RF9 and try a 'ahrstrim'. The HUD goes mad without this change

Reply:
> I think my preferred fix would be to check if trim or cal is running and in that case do a early return on 3709, with a nice comment explaining the loop is stuck

I tried that and it doesn't work. As soon as the cal is over we then run the time comparison, and it is over 1s even though cal is now finished, so we still show the wrong attitude briefly, and HUD still goes crazy.
We'd need a "did we run a cal in the last 2s" call.